### PR TITLE
KRKPD-987: floors should reference Floors Module

### PR DESCRIPTION
**Pull request by nickllerandi**
## Type of change

- [x] Bugfix

## Description of change
- Previous code was referencing implementation specific to [analytics adapters](https://docs.prebid.org/dev-docs/modules/floors.html#bid-adapter-interface:~:text=Example%20getFloor()%20scenarios-,Analytics%20Adapter%20Interface,-Prebid%20Server%20Interface). This changes code to use [bid adapter implementation](https://docs.prebid.org/dev-docs/modules/floors.html#bid-adapter-interface:~:text=3%20%2D%20Schema%202-,Bid%20Adapter%20Interface,-getFloor()%20Response). 
- `getFloor({mediaType: '*',size: '*'` utilizes Prebid's smart rule selection to choose most appropriate pub-specific price rule based on adunit configured mediatype and size

Screenshot 2024-03-07 at 3 06 16 PM

Open Questions:
- [ ] consider native (remove code? or add as supported mediatype?)

### CODE REVIEW

**Comment by nickllerandi** [comment]
[Initial commit](https://github.com/KargoGlobal/Prebid.js/pull/26/commits/eb6747feea69471721ee2d117c5a41add0ec0822) was a more straightforward implementation - but I think we have some options here.

File: modules/kargoBidAdapter.js
Comment (nickllerandi) [inline modules/kargoBidAdapter.js:491]: should native be here if it's not a supported mediaType?

File: modules/kargoBidAdapter.js
Comment (nickllerandi) [inline modules/kargoBidAdapter.js:495]: `mediatype/size:"*"` will leverage Prebid's [smart rule selection](https://docs.prebid.org/dev-docs/modules/floors.html#analytics-adapter-interface:~:text=smart%20rule%20selection)

File: modules/kargoBidAdapter.js
Comment (juliangan07) [inline modules/kargoBidAdapter.js]: We support native through Kargo Ad Tag so the ad markup submitted would be considered banner, there isn't any news about us supporting native responses directly, so i'm incline to say it's fine to leave out.

File: modules/kargoBidAdapter.js
Comment (juliangan07) [reply modules/kargoBidAdapter.js:491]: Probably okay to have it here, while we don't have any pubs that sends us native requests today, but this way, we'd at least be able to receive if there's an initiative to start supporting it.

**Review by juliangan07 — APPROVED** [review]
Looks good! Just responded with some comments about native.

**Comment by nickllerandi** [comment]
Going to align with @sj1815 first before merging, as merging into master now would then be reflected in [this](https://github.com/prebid/Prebid.js/pull/11153) PR as well. Probably best to wait for the creativeID stuff to be approved/merged into Prebid/Prebid so we're not stepping on toes.

**Review by elixsantana — APPROVED** [review]

**Review by jsadwith — APPROVED** [review]
Looks great - thanks Nick!

**Pull request merged**
